### PR TITLE
Fixes for Azure Automation module 

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
This PR contains the following fixes:

- Fixing odata filter for getting a webhook by runbook name
- Fixing JobSchedule by setting the correct JobScheduleId
- Fixing exception handling for source control cmdlets
- Adding logic to ExecuteCmdlet, so that when a ErrorResponseException is thrown, we check to see if ErrorResponseException.Body.Message is available. If not, we try to extracted from error message from ErrorResponseException.Response.Content, and we rethrow again using the extracted message. With this fix, the correct error message for invalid ResourceGroupName and AutomationAccountName is displayed to the user.

### CODE REVIEW

File: src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationNestedErrorResponseMessage.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        public AzureAutomationErrorInfo Error { get; set; }
+
+        // Initializes a new instance of the class.
+        public AzureAutomationNestedErrorResponseMessage(string code, string message)
+        {
+            this.Error = new AzureAutomationErrorInfo { Code = code, Message = message };
+        }
+
+        /// The error information.
+        public class AzureAutomationErrorInfo
```
Comment (mirichmo) [inline src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationNestedErrorResponseMessage.cs]: This is not necessary if you use AzureAutomationErrorResponseMessage

File: src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationErrorResponseMessage.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Automation.Common
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// The error response message.
+    /// </summary>
+    public class AzureAutomationErrorResponseMessage
```
Comment (mirichmo) [inline src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationErrorResponseMessage.cs:22]: How about naming it AzureAutomationErrorInfo?

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/AzureAutomationBaseCmdlet.cs [hunk 3]
```diff
@@ -131,29 +168,24 @@ protected bool GenerateCmdletOutput(IEnumerable<object> results)
             return ret;
         }
 
-        private string ParseErrorMessage(string errorMessage)
+        // This function determines if the given value is a Json object.
+        private bool IsValidJsonObject(string value)
```
Comment (mirichmo) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/AzureAutomationBaseCmdlet.cs]: Is this really necessary? It seems like extra work. How about just attempting to parse and handling errors?

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationNestedErrorResponseMessage.cs (hunk 1 above)
Comment (Francisco-Gamino) [reply src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationNestedErrorResponseMessage.cs]: Good point. Fixed.

File: src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationErrorResponseMessage.cs (hunk 2 above)
Comment (Francisco-Gamino) [reply src/ResourceManager/Automation/Commands.Automation/Common/AzureAutomationErrorResponseMessage.cs:22]: Sounds good. Fixed.

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/AzureAutomationBaseCmdlet.cs (hunk 3 above)
Comment (Francisco-Gamino) [reply src/ResourceManager/Automation/Commands.Automation/Cmdlet/AzureAutomationBaseCmdlet.cs]: I agree. I will go ahead and remove the code that checks if it is a valid Json. In ParseJson, if we are not able to extract the error message, we just return null.

**Review by vrdmr — APPROVED** [review]

**Review by mirichmo — APPROVED** [review]

**Pull request merged**